setup.py

```diff
@@ -26,6 +26,37 @@ def read(filename):
         return f.read()
 
 
+def read_reqs(filename):
+    return read(filename).strip().splitlines()
+
+
+install_requires = read_reqs('etc/requirements.txt')
+
+
+def extras_require():
+    sql = read_reqs('etc/requirements_sql.txt')
+
+    aws = read_reqs('etc/requirements_aws.txt')
+    bcolz = read_reqs('etc/requirements_bcolz.txt')
+    bokeh = read_reqs('etc/requirements_bokeh.txt')
+    dask = read_reqs('etc/requirements_dask.txt')
+    h5py = read_reqs('etc/requirements_h5py.txt')
+    mongo = read_reqs('etc/requirements_mongo.txt')
+    mysql = sql + read_reqs('etc/requirements_mysql.txt')
+    postgres = sql + read_reqs('etc/requirements_postgres.txt')
+    pytables = read_reqs('etc/requirements_pytables.txt')
+    sas = read_reqs('etc/requirements_sas.txt')
+    ssh = read_reqs('etc/requirements_ssh.txt')
+    test = read_reqs('etc/requirements_test.txt')
+
+    extras = locals()
+    extras['all'] = list(chain.from_iterable(extras.values()))
+    # write the ci target after collecting all of the others so that the ci
+    # frozen dependencies don't trample the less strict versions.
```

Comment:
Instead of relying on no unexpected locals being introduced, what do you think of:
```python
def extras_require():
    extras = {
        req: read_reqs('etc/requirements_{req}'.format(req=req)) 
        for req in {
          'sql', 'aws', 'bcolz', 'bokeh', 'dask', 'h5py', 'mongo', 'mysql', 'postgres',
          'pytables', 'sas', 'ssh', 'test', 'ci',
        }
    }
    extras['mysql'] += extras['sql']
    extras['postgresql'] += extras['sql']
    return extras
```